content/docs/user-guide/dvc-files.md

```diff
@@ -240,14 +248,245 @@ stages:
 ```
 
 Stage commands are listed again in `dvc.lock`, in order to know when their
-definitions change in the `dvc.yaml` file.
+definitions change in `dvc.yaml`.
 
 Regular <abbr>dependencies</abbr> and all kinds of <abbr>outputs</abbr>
 (including [metrics](/doc/command-reference/metrics) and
 [plots](/doc/command-reference/plots) files) are also listed (per stage) in
-`dvc.lock`, but with an additional field to store the hash value of each file or
-directory tracked by DVC. Specifically: `md5`, `etag`, or `checksum` (same as in
-`deps` and `outs` entries of `.dvc` files).
+`dvc.lock`, but with an additional field with a hash of their last known
+contents. Specifically: `md5`, `etag`, or `checksum` are used (same as in `deps`
+and `outs` entries of `.dvc` files).
+
+Full <abbr>parameter dependencies</abbr> (key and value) are listed too (under
+`params`), grouped by parameters file. And in the case of
+templated/parameterized `dvc.yaml` files (see next section), their actual values
+are substituted into the `dvc.lock` YAML structure.
+
+### Parameterize `dvc.yaml`
+
+`dvc.yaml` supports a templating format to reuse values from different sources
+in the YAML structure itself. The sources can be external
+[parameters files](/doc/command-reference/params), or internal `vars`, as
+detailed below.
+
+Let's say we have `params.yaml` (default params file) with the following
+contents:
+
+```yaml
+models:
+  us:
+    threshold: 10
+    filename: 'model-us.hdf5'
+```
+
+Those values can be used anywhere in `dvc.yaml` with the `${}` _substitution
+expression_:
+
+<!-- prettier-ignore-start -->
+```yaml
+stages:
+  build-us:
+    cmd: >-
+      python train.py
+      --thresh ${models.us.threshold}
+      --out ${models.us.filename}
+    outs:
+      - ${models.us.filename}:
+        cache: true
+```
+<!-- prettier-ignore-end -->
+
+DVC will track simple param values (numbers, strings, etc.) used in `${}` (they
+will be listed by `dvc params diff`).
+
+Alternatively, values for substitution can be listed as top-level `vars` like
+this:
+
+```yaml
+vars:
+  - models:
+      us:
+        threshold: 10
+  - desc: 'Reusable description'
+
+stages:
+  build-us:
+    desc: ${desc}
+    cmd: python train.py --thresh ${models.us.threshold}
+```
+
+> DVC merges external params and `vars` as long as there are no leaf node
+> collisions (so the two examples above can't be used simultaneously). For
+> example, `{"grp": {"a": 1}}` can be merged with `{"grp": {"b": 2}}`, but not
+> with `{"grp": {"a": 7}}`.
+
+⚠️ Note that doesn't track values from `vars` like parameters (ignored by
```

Comment:
it's not needed I think

Reply:
@skshetry mentions when values are tracked and when they're not in https://github.com/iterative/dvc/wiki/Parametrization so I assumed it was important. I personally think it can be relevant to (be able to) know this, but no strong opinion (b/c on the other hand it's kind of expected/intuitive). Up to you, please lmk 🙂

Reply:
> `... doesn't track values from `vars` like parameters`

p.s. for now I made it into a grayed note.